### PR TITLE
Implement sorting on top level abstract types query fields

### PR DESCRIPTION
**Pull request by a-alle**
# Description

This PR adds sorting and pagination capabilities to top-level interface & union queries.
Depends on https://github.com/neo4j/graphql/pull/4078

## Complexity

Complexity: Low

# Issue

> **Note**
>
>  Please link to the GitHub issue(s) in which the proposal for this work was discussed
>  
>  To link to multiple issues, use full syntax for each, for example `Closes #1, closes #2, closes #3`

Closes 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-4089.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 44 | 91 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 114 | 127 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 62 | 78 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 4912 | 5039 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 4400 | 4477 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 66 | 91 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 77 | 96 | 2174 |
|  aggregations.DeeplyNestedCount | 11062443 | 11062443 | 6518 | 7076 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 207 | 265 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 76 | 134 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 293 | 344 | 3003 |
|  connections.Connection | 14082 | 14082 | 85 | 115 | 2174 |
|  connections.NestedConnection | 41118 | 41118 | 148 | 243 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 91 | 102 | 1040 |
|  connections.ConnectionWithSortAndCypher | 16325 | 16325 | 124 | 205 | 2174 |
|  create.SimpleMutation | 7 | 7 | 77 | 113 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 137 | 48 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 451 | 372 | 1040 |
|  delete.NestedDeleteInUpdate | 19542 | 19542 | 349 | 295 | 1183 |
|  2871.NestedRelationshipFilter | 23066 | 23066 | 135 | 114 | 4414 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 70 | 72 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 123 | 195 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 59 | 69 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 83 | 120 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 31 | 39 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 64 | 76 | 2174 |
|  query.QueryWhere | 9711 | 9711 | 55 | 63 | 2166 |
|  query.SimpleQueryWithNestedWhere | 9889 | 9889 | 77 | 98 | 2166 |
|  query.Nested | 10096041 | 10096041 | 12308 | 12473 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 12294 | 12765 | 2004000 |
|  query.OrFilterOnRelationships | 42859 | 42898 | 290 | 391 | 1785 |
|  query.OrFilterOnRelationshipsAndNested | 36587 | 36129 | 300 | 410 | 1785 |
|  query.QueryWithNestedIn | 13561 | 13580 | 83 | 130 | 1269 |
|  query.NestedConnectionWhere | 9834 | 9834 | 83 | 92 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9851 | 9851 | 108 | 144 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18996 | 18996 | 203 | 246 | 1552 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 310 | 430 | 1134 |
|  query.Fulltext | 80 | 80 | 51 | 62 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 72 | 79 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 90 | 114 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14464 | 14464 | 261 | 152 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 162 | 90 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 150 | 143 | 4516 |
|  sorting.SortWithTopLevelCypher | 3396 | 3396 | 83 | 105 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 85 | 95 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 82 | 107 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 407 | 453 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 405 | 534 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 183 | 179 | 2002 |

</details>
        
Old Schema Generation: 1:02.547 (m:ss.mmm)
Schema Generation: 1:06.674 (m:ss.mmm)
Old Subgraph Schema Generation: 1:15.554 (m:ss.mmm)
Subgraph Schema Generation: 1:20.268 (m:ss.mmm)

File: packages/graphql/tests/tck/union.test.ts [hunk 1]
```diff
@@ -159,6 +159,92 @@ describe("Cypher Union", () => {
         `);
     });
 
+    test("Read union top-level with sort and filter", async () => {
```
[the remark this answers is not in the document]
Comment (a-alle) [reply packages/graphql/tests/tck/union.test.ts:162]: I get the point, however in this particular case sort and limit do not influence each other and are at the same level. In my opinion having them in the same test does not make debugging harder. Let me know what you think!

File: packages/graphql/tests/tck/sort.test.ts [hunk 2]
```diff
@@ -351,4 +351,390 @@ describe("Cypher sort tests", () => {
...
+                }
+            }
+        `;
+
+        const result = await translateQuery(neoSchema, query);
+
+        expect(formatCypher(result.cypher)).toMatchInlineSnapshot(`
+            "CALL {
+                MATCH (this0:SomeNodeType)
+                WHERE NOT (this0.somethingElse = $param0)
```
Comment (MacondoExpress) [inline packages/graphql/tests/tck/sort.test.ts:414]: 👀

**Review by MacondoExpress — DISMISSED** [review]
Looks nice!

**Pull request merged**